components/layout_2020/display_list/stacking_context.rs

```diff
@@ -708,13 +710,35 @@ impl BoxFragment {
         builder.current_space_and_clip.spatial_id = builder.nearest_reference_frame;
     }
 
+    fn build_clip_frame_if_necessary(
+        &self,
+        builder: &mut StackingContextBuilder,
+        containing_block_info: &ContainingBlockInfo,
+    ) {
+        let clip = self.style.get_effects().clip;
+        if let ClipRectOrAuto::Rect(r) = clip {
```

Comment:
https://drafts.csswg.org/css2/#clipping
> The clip property applies only to absolutely positioned elements. 

So this code should also check for `position: absolute` or `position: fixed`.